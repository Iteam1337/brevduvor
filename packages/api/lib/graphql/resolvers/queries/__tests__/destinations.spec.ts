export {} // TypeScript diagnostics
const { ApolloServer } = require('apollo-server-express')
const { createTestClient } = require('apollo-server-testing')
const { gql } = require('apollo-server-express')
const { allDestinations } = require('../index')
const { serverConfig } = require('../../../../../index')
const { query } = createTestClient(
  new ApolloServer({
    typeDefs: serverConfig.typeDefs,
    schemaDirectives: serverConfig.schemaDirectives,
    resolvers: serverConfig.resolvers,
  })
)

const dbResponse = [
  {
    alias: 'Storuman',
    lat: 65.0908,
    lon: 17.1075,
  },
  {
    alias: 'Kvikkjokk',
    lat: 66.9501,
    lon: 17.7086,
  },
]

jest.mock('../../../../services/destinations', () => ({
  getDestinations: jest.fn(() => dbResponse),
}))

describe('destinations', () => {
  describe('allDestinations', () => {
    it('exists', () => {
      expect(allDestinations).toBeDefined()
    })

    it('matches snapshot', async () => {
      const ALL_DEST = gql`
        query {
          allDestinations {
            alias
            lat
            lon
          }
        }
      `

      const res = await query({ query: ALL_DEST })

      expect(res.data).toMatchSnapshot()
    })
  })
})
