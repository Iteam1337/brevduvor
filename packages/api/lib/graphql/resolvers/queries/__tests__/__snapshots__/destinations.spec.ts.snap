// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`destinations allDestinations matches snapshot 1`] = `
Object {
  "allDestinations": Array [
    Object {
      "alias": "Storuman",
      "lat": 65.0908,
      "lon": 17.1075,
    },
    Object {
      "alias": "Kvikkjokk",
      "lat": 66.9501,
      "lon": 17.7086,
    },
  ],
}
`;
